contrib/lib/src/lib.rs

```diff
@@ -43,12 +42,31 @@
 #[allow(unused_imports)] #[macro_use] extern crate log;
 #[allow(unused_imports)] #[macro_use] extern crate rocket;
 
-#[cfg(feature="json")] #[macro_use] pub mod json;
-#[cfg(feature="serve")] pub mod serve;
-#[cfg(feature="msgpack")] pub mod msgpack;
-#[cfg(feature="templates")] pub mod templates;
-#[cfg(feature="uuid")] pub mod uuid;
-#[cfg(feature="databases")] pub mod databases;
+#[cfg(feature = "json")]
+#[macro_use]
+pub mod json;
+#[cfg(feature = "databases")]
+pub mod databases;
+#[cfg(feature = "msgpack")]
+pub mod msgpack;
+#[cfg(feature = "serve")]
+pub mod serve;
+#[cfg(feature = "templates")]
+pub mod templates;
+#[cfg(feature = "uuid")]
+pub mod uuid;
+
+#[cfg(feature = "databases")]
+extern crate rocket_contrib_codegen;
+#[cfg(feature = "databases")]
+#[doc(hidden)]
+pub use rocket_contrib_codegen::*;
+
+#[cfg(any(feature = "brotli_compression", feature = "gzip_compression"))]
+pub mod compression;
+
+#[cfg(feature = "brotli_compression")]
+extern crate brotli;
```

Comment:
This `extern crate` really should be in `compression.rs` for consistency with the rest of `contrib`. I'll try to get it to work again before this is merged.